src/components/UpdateButton/UpdateVisualizationContainer.js

```diff
@@ -8,57 +8,27 @@ import {
 import { acSetShowExpandedLayoutPanel } from '../../actions/ui.js'
 import { genericClientError } from '../../modules/error.js'
 import { validateLayout } from '../../modules/layoutValidation.js'
-import { sGetCurrent, sGetCurrentFromUi } from '../../reducers/current.js'
+import { sGetCurrentFromUi } from '../../reducers/current.js'
+
+const UpdateVisualizationContainer = ({ renderComponent }) => {
+    const dispatch = useDispatch()
+    const currentFromUi = useSelector(sGetCurrentFromUi)
 
-const UpdateVisualizationContainer = ({
-    renderComponent,
-    getCurrentFromUi,
-    onUpdate,
-    setLoadError,
-    clearLoadError,
-    onLoadingStart,
-    hideLayoutPanel,
-}) => {
     const onClick = () => {
         try {
-            validateLayout(getCurrentFromUi())
-            clearLoadError()
-            onUpdate()
-            onLoadingStart()
+            validateLayout(currentFromUi)
```

Comment:
why not use the selector directly?
```suggestion
            validateLayout(useSelector(sGetCurrentFromUi))
```

Reply:
because I didn't think about it.